From deaae6eddef5186bfd0e42e2c3ced39e26afa4d6 Mon Sep 17 00:00:00 2001
From: "Joseph C. Sible" <josephcsible@users.noreply.github.com>
Date: Sat, 31 Aug 2019 21:19:37 -0400
Subject: [PATCH] Remove fail from default_typeclass test

This was added when we copied in StateT from MTL, but we don't actually
use it. Since the MonadFail Proposal was finished in GHC 8.8.1, it now
breaks compiling, so remove it.
---
 tests/default_typeclass.x | 1 -
 1 file changed, 1 deletion(-)

diff --git a/tests/default_typeclass.x b/tests/default_typeclass.x
index fb06684..5491e28 100644
--- a/tests/default_typeclass.x
+++ b/tests/default_typeclass.x
@@ -285,7 +285,6 @@ instance (Monad m) => Monad (StateT s m) where
     m >>= k  = StateT $ \s -> do
         (a, s') <- runStateT m s
         runStateT (k a) s'
-    fail str = StateT $ \_ -> fail str
 
 -- | Fetch the current value of the state within the monad.
 get' :: (Monad m) => StateT s m s
-- 
2.16.1

